dataflow/src/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -74,6 +74,18 @@
     protected IdentityHashMap<Block, S> elseStores;
 
     /**
+     * Number of times every block has been analyzed since the last time widening was applied. Null,
+     * if maxCountBeforeWidening is -1 which implies widening isn't using for this analysis.
+     */
+    protected IdentityHashMap<Block, Integer> blockCount;
+
+    /**
+     * Number of times a block can be analyzed before widening. -1 implies that widening shouldn't
+     * be used.
+     */
+    protected int maxCountBeforeWidening = -1;
```

Comment:
I found how this field gets set confusing. It first is -1 here and then gets over-written in the subclass.
How about making the field final, adding a new constructor that takes a parameter for the field, and have the current constructors call the new constructor with -1?